### PR TITLE
[Model] HeCo

### PR DESCRIPTION
**Pull request by tanjjjjrr**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by dddg617** [comment]
1. Lack the `readme.md` in the example file, which includes the model performance.
2. Have you tested the example? I think there should be some bugs, e.g. you do not modify the `__init__.py`.
3. There should be docstring in the dataset, your layer/conv, and model.

File: gammagl/datasets/acm4heco.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+import tensorlayerx as tlx
+import shutil
+import errno
+from gammagl.data import (HeteroGraph, InMemoryDataset, extract_zip)
+from sklearn.preprocessing import OneHotEncoder
+import requests
+import io
+import zipfile
+from gammagl.data import download_url
+r"""The heterogeneous ACM dataset from the `"Self-supervised Heterogeneous Graph Neural Network with
```
Comment (dddg617) [inline gammagl/datasets/acm4heco.py:22]: Error location! Refer to the "document generation guide" and test on your local laptop. This docstring will not generate correctly.

File: gammagl/models/HeCo.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import tensorlayerx as tlx
+import tensorlayerx.nn as nn
+from layers.attention.HeCo_encoder import Mp_encoder
+from layers.attention.HeCo_encoder import Sc_encoder
+r"""HeCo proposed in `"Self-supervised Heterogeneous Graph Neural Network with Co-contrastive Learning"
```
Comment (dddg617) [inline gammagl/models/HeCo.py:6]: Ditto.

File: examples/HeCo/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+|    40                   | 87.61±0.5| 83.4±0.3 | 85.4±0.1 | 85.6±0.2   |
+|    60                   | 89.04±0.5| 81.4±0.4 | 83.4±0.4 | 83.3±0.3   |
+
+- Mi-F1
+
+| number of train_labels  | Paper    | Our(tf)  | Our(pd)  | Our(torch) |
+|-------------------------|----------|----------|----------|------------|
+|    20                   | 88.13±0.8| 79.1±0.4 | 80.44±0.8| 80.4±0.7   |
+|    40                   | 87.45±0.5| 83.4±0.3 | 85.43±0.1| 85.4±0.2   |
+|    60                   | 88.71±0.5| 79.4±0.4 | 82.2±0.5 | 82.4±0.6   |
```
Comment (dddg617) [inline examples/HeCo/README.md:29]: @gyzhou2000 If this performance is far beyond expected, as the performance is about 5% lower than the original paper.

File: gammagl/models/HeCo.py (hunk 2 above)
Comment (tanjjjjrr) [reply gammagl/models/HeCo.py:6]: sorry for that! I have changed Mp_encoder and Sc_encoder files inside, it has no error on my laptop.

**Comment by tanjjjjrr** [comment]
yep, everything goes right now. I have forgotten to take the discarded backend mindspore before, which turns out the code was wrong.

File: gammagl/models/HeCo.py (hunk 2 above)
Comment (dddg617) [reply gammagl/models/HeCo.py:6]: You tried the document generation, not just running the code?

Comment (tanjjjjrr) [reply gammagl/models/HeCo.py:6]: I might have tried all link relations of the code on my laptop(including data generation etc, while I was running HeCo_trainer.py)and all works well. Could you please give me some hints about the document generation?

**Pull request merged**